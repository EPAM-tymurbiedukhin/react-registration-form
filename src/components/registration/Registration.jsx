import React from 'react';

import MainInput from '../mainInput';

import {
  INVALID_NAME,
  INVALID_PASSWORD,
  INVALID_EMAIL,
} from '../../constants/validation-messages';
import {
  EMAIL_INPUT,
  NAME_INPUT,
  PASSWORD_INPUT,
} from '../../constants/validation-rules';

export default ({handleSubmit, handleInput}) => {
  return (
    <form onSubmit={handleSubmit} className="form">
      <MainInput
        name="first_name"
        formName="registration"
        type="text"
        placeholder="First Name"
        isRequired={true}
        validation={NAME_INPUT}
        errorMessage={INVALID_NAME}
        onChange={handleInput}
      />
      <MainInput
        name="last_name"
        formName="registration"
        type="text"
        placeholder="Last Name"
        validation={NAME_INPUT}
        errorMessage={INVALID_NAME}
        onChange={handleInput}
      />
      <MainInput
        name="email"
        formName="registration"
        type="email"
        placeholder="Email"
        isRequired={true}
        validation={EMAIL_INPUT}
        errorMessage={INVALID_EMAIL}
        onChange={handleInput}
      />
      <MainInput
        name="password"
        formName="registration"
        type="password"
        placeholder="Password"
        isRequired={true}
        validation={PASSWORD_INPUT}
        errorMessage={INVALID_PASSWORD}
        onChange={handleInput}
      />
      <MainInput
        name="repeat_password"
        formName="registration"
        type="password"
        placeholder="Repeat password"
        isRequired={true}
        validation={PASSWORD_INPUT}
        errorMessage={INVALID_PASSWORD}
        onChange={handleInput}
      />
      <button className="waves-effect waves-light btn">submit</button>
    </form>
  );
};
