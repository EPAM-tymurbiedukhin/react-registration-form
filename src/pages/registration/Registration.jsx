import React from 'react';
import { useDispatch } from 'react-redux';

import MainWrapper from '../../components/mainWrapper';
import Breadcrumbs from '../../components/navigations';
import Registration from '../../components/registration';

import {
  setRegistrationField,
  submitRegistrationForm,
} from '../../features/registration/actions';

export default () => {
  const dispatch = useDispatch();

  const handleInput = ({ name, value, isRequired, errorMessage, isValid }) => {
    dispatch(setRegistrationField({ name, value, isRequired, errorMessage, isValid }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    dispatch(submitRegistrationForm());
  };

  return (
    <MainWrapper>
      <Registration handleSubmit={handleSubmit} handleInput={handleInput} />
      <Breadcrumbs />
    </MainWrapper>
  );
};
