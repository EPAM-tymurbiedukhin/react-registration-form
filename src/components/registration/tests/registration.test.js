import React from 'react';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import enzyme, { shallow, mount, render } from 'enzyme';

import Registration from '../Registration';

import {
  stubRegistrationProps,
  defaultValue,
  onChange,
  registrationLinks,
} from './stubProps';

// const mockStore = createMockStore();

jest.mock('react-redux', () => ({
  useSelector: jest.fn(() => 'default value'), // defaultValue
}));

const setUp = (props) =>
  mount(
    // <Provider store={mockStore()}>
    <Registration {...props} />
    //</Provider>
  );

describe('<Registration />', () => {
  let component;
  let form;

  beforeEach(() => {
    component = setUp(stubRegistrationProps);
    form = component.find('form');
  });

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('should form has correct class', () => {
    expect(form.hasClass('form')).toBeTruthy();
  });

  it('should call handleSubmit on submit form', () => {
    form.simulate('submit');
    expect(stubRegistrationProps.handleSubmit).toHaveBeenCalled();
  });

  it('should all inputs render correctly', () => {
    component.find('input').forEach((node, i) => {
      const nodeProps = node.props();
      const {name, type, placeholder } = registrationLinks[i];

      expect(nodeProps.name).toBe(name);
      expect(nodeProps.type).toBe(type);
      expect(nodeProps.id).toBe(name);
      expect(nodeProps.value).toBe(defaultValue);
      expect(nodeProps.placeholder).toBe(placeholder);
      node.simulate('change');
      expect(stubRegistrationProps.handleInput).toHaveBeenCalled();
    });
  });

  it('should call submitForm on button click', () => {
    const button = component.find('button');
    button.simulate('click');
    expect(stubRegistrationProps.handleSubmit).toHaveBeenCalled();
  })
});
