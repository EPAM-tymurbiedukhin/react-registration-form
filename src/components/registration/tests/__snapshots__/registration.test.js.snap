// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> should render correctly 1`] = `"<form class=\\"form\\"><div class=\\"input-field col s12\\"><input name=\\"first_name\\" type=\\"text\\" id=\\"first_name\\" required=\\"\\" placeholder=\\"First Name\\" class=\\"\\" value=\\"default value\\"><label for=\\"first_name\\"></label></div><div class=\\"input-field col s12\\"><input name=\\"last_name\\" type=\\"text\\" id=\\"last_name\\" placeholder=\\"Last Name\\" class=\\"\\" value=\\"default value\\"><label for=\\"last_name\\"></label></div><div class=\\"input-field col s12\\"><input name=\\"email\\" type=\\"email\\" id=\\"email\\" required=\\"\\" placeholder=\\"Email\\" class=\\"\\" value=\\"default value\\"><label for=\\"email\\"></label></div><div class=\\"input-field col s12\\"><input name=\\"password\\" type=\\"password\\" id=\\"password\\" required=\\"\\" placeholder=\\"Password\\" class=\\"\\" value=\\"default value\\"><label for=\\"password\\"></label></div><div class=\\"input-field col s12\\"><input name=\\"repeat_password\\" type=\\"password\\" id=\\"repeat_password\\" required=\\"\\" placeholder=\\"Repeat password\\" class=\\"\\" value=\\"default value\\"><label for=\\"repeat_password\\"></label></div><button class=\\"waves-effect waves-light btn\\">submit</button></form>"`;
